### ISSUE TITLE
Create a test CONVNET with MNIST

### ISSUE THREAD

**Issue by leonardoaraujosantos** [leonardoaraujosantos/DLMatFramework#18]
## Introduction
In order to validate all the layers together, create a small network
CONV-->BN-->ReLU-->M_POOL-->CONV-->BN-->ReLU-->M_POOL-->FC-->BN-->Drop-->FC-->Softmax
Using Muli-class cross entropy

Train with a subset of MNIST and check how is going.

## Creating network
``` matlab
%% Create network
layers = LayerContainer();    
layers <= struct('name','ImageIn','type','input','rows',28,'cols',28,'depth',1, 'batchsize',1);
layers <= struct('name','CONV1','type','conv', 'kh',5,'kw',5,'stride',1,'pad',2,'num_output', 32); 
layers <= struct('name','SBN_1','type','sp_batchnorm','eps',1e-5, 'momentum', 0.9);
layers <= struct('name','Relu_1','type','relu');
layers <= struct('name','MP1','type','maxpool', 'kh',2, 'kw',2, 'stride',2); 
layers <= struct('name','CONV2','type','conv', 'kh',5,'kw',5,'stride',1,'pad',2,'num_output', 64); 
layers <= struct('name','SBN_2','type','sp_batchnorm','eps',1e-5, 'momentum', 0.9);
layers <= struct('name','Relu_2','type','relu');
layers <= struct('name','MP2','type','maxpool', 'kh',2, 'kw',2, 'stride',2); 
layers <= struct('name','FC_3','type','fc', 'num_output',1024);
layers <= struct('name','BN_3','type','batchnorm','eps',1e-5, 'momentum', 0.9);
layers <= struct('name','Relu_3','type','relu');
layers <= struct('name','DRP_1','type','dropout','prob',0.5);
layers <= struct('name','FC_4','type','fc','num_output',data.GetNumClasses());
layers <= struct('name','Softmax','type','softmax');

% Create DeepLearningModel instance
net = DeepLearningModel(layers, LossFactory.GetLoss('multi_class_cross_entropy'));


%% Create solver and train
solver = Solver(net, data, 'sgd',containers.Map({'learning_rate', 'L2_reg'}, {0.1, 0}));
solver.SetBatchSize(256);
solver.SetEpochs(10);
solver.PrintEvery(10);
solver.Train();
```

### Results
99% Validation accuracy

### With batchnorm
![losswithbatchnorm](https://cloud.githubusercontent.com/assets/898383/22701379/6f00e5a2-ed55-11e6-93de-42af58f75fef.png)

### Batchnorm disabled
![withoutbatchnorm_mnist_conv](https://cloud.githubusercontent.com/assets/898383/22701412/7edc7838-ed55-11e6-9252-d5579ffbe9c2.png)

**Comment by leonardoaraujosantos** [comment]
By using the complete MNIST the validation accuracy for this network was 99%, which is what you get on tensorflow, so we're good for now.
